int32 closeWidth
float32 targetRotationAngle
string modelType
string objectType